SlideServer.py

```diff
@@ -150,3 +150,11 @@ def singleThumb(filepath):
 @app.route("/data/many/<filepathlist>", methods=['GET'])
 def multiSlide(filepathlist):
     return json.dumps(dev_utils.getMetadataList(json.loads(filepathlist), app.config['UPLOAD_FOLDER']))
+
+@app.route("/getSlide/<image_name>")
+def getSlide(image_name):
+    if(os.path.isfile("/images/"+image_name)):
+        return flask.send_from_directory(app.config["UPLOAD_FOLDER"], filename=image_name, as_attachment=True)
```

Comment:
I think this idea is workable, since we can ask users/admins concerned about having private slides to mount them in a different directory, or otherwise not mount them to this container. For the former:
It looks like isfile can be fooled by path traversal
```>>> os.path.isfile("/home/ryan/Documents/Distro/images" + "/../README.md")
True```
Can you confirm that this doesn't work with send_from_directory? (e.g. put a slide file somewhere in the container outside of /images and its children, and see if you can force the server into downloading it)

Reply:
As to my knowledge `send_from_directory()` is secure for passing file names.
`send_file()` should not be used in such cases so I used this.

Reply:
This may be used to get inside of the `if` statement but I don't think that `send_from_directory()` will fall into this.

Reply:
I think you're right.